File: dp_labs_and_solutions/dp_labs_and_solutions/dp_cpp_labs_and_solutions/interpreter/lab/unary_minus.cpp
// unary_minus.cpp

#include <iostream>
#include "dp_exception.h"
class expression;
#include "unary.h"
#include "unary_minus.h"

unary_minus::unary_minus(expression * a_operand1)
: unary(a_operand1)
{
}

void unary_minus::output(std::ostream & a_ostream) const
{
	output_prefix(a_ostream, "-");
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_java_labs_and_solutions/interpreter/solution/UnaryMinus.java
// UnaryMinus.java

import java.io.PrintWriter;

public class UnaryMinus extends Unary
{
	public UnaryMinus(Expression aOperand1)
	{
		super(aOperand1);
	}

	public double getValue(DpMap<String, Double> aMap)
	{
		if (getOperand1() == null)
		{
			throw new DpException("void UnaryMinus.getValue(DpMap<String, Double> aMap) -- getOperand1() == null.");
		}
		double aValue1 = getOperand1().getValue(aMap);

		double result = - aValue1;

		return result;
	}

	public void output(PrintWriter aPrintWriter)
	{
		outputPrefix(aPrintWriter, "-");
	}
}
